### PR TITLE
BUG: display.precision of negative complex numbers

### LINKED ISSUE #27484: IndexError in repr of series objects containing complex numbers with negative imaginary parts
#### Code Sample, a copy-pastable example if possible

```python
from pandas import Series

print(Series([-1j]))

```
#### Problem description

This raises the following error:

```
Traceback (most recent call last):
  File "foo.py", line 3, in <module>
    print(Series([-1j]))
  File "/home/david/.pyenv/versions/3.7.4/lib/python3.7/site-packages/pandas/core/series.py", line 1611, in __repr__
    length=show_dimensions,
  File "/home/david/.pyenv/versions/3.7.4/lib/python3.7/site-packages/pandas/core/series.py", line 1677, in to_string
    result = formatter.to_string()
  File "/home/david/.pyenv/versions/3.7.4/lib/python3.7/site-packages/pandas/io/formats/format.py", line 312, in to_string
    fmt_values = self._get_formatted_values()
  File "/home/david/.pyenv/versions/3.7.4/lib/python3.7/site-packages/pandas/io/formats/format.py", line 299, in _get_formatted_values
    values_to_format, None, float_format=self.float_format, na_rep=self.na_rep
  File "/home/david/.pyenv/versions/3.7.4/lib/python3.7/site-packages/pandas/io/formats/format.py", line 1032, in format_array
    return fmt_obj.get_result()
  File "/home/david/.pyenv/versions/3.7.4/lib/python3.7/site-packages/pandas/io/formats/format.py", line 1063, in get_result
    fmt_values = self._format_strings()
  File "/home/david/.pyenv/versions/3.7.4/lib/python3.7/site-packages/pandas/io/formats/format.py", line 1288, in _format_strings
    return list(self.get_result_as_array())
  File "/home/david/.pyenv/versions/3.7.4/lib/python3.7/site-packages/pandas/io/formats/format.py", line 1252, in get_result_as_array
    formatted_values = format_values_with(float_format)
  File "/home/david/.pyenv/versions/3.7.4/lib/python3.7/site-packages/pandas/io/formats/format.py", line 1234, in format_values_with
    return _trim_zeros_complex(values, na_rep)
  File "/home/david/.pyenv/versions/3.7.4/lib/python3.7/site-packages/pandas/io/formats/format.py", line 1597, in _trim_zeros_complex
    return ["".join(separate_and_trim(x, na_rep)) for x in str_complexes]
  File "/home/david/.pyenv/versions/3.7.4/lib/python3.7/site-packages/pandas/io/formats/format.py", line 1597, in <listcomp>
    return ["".join(separate_and_trim(x, na_rep)) for x in str_complexes]
  File "/home/david/.pyenv/versions/3.7.4/lib/python3.7/site-packages/pandas/io/formats/format.py", line 1594, in separate_and_trim
    + ["j"]
IndexError: list index out of range
```

#### Expected Output

This should print something like the following:

```
0    0.0-1.0j
dtype: complex128
```

#### Output of ``pd.show_versions()``

<details>

[paste the output of ``pd.show_versions()`` here below this line]

INSTALLED VERSIONS
------------------
commit           : None
python           : 3.7.4.final.0
python-bits      : 64
OS               : Linux
OS-release       : 4.4.0-17134-Microsoft
machine          : x86_64
processor        : x86_64
byteorder        : little
LC_ALL           : None
LANG             : en_US.UTF-8
LOCALE           : en_US.UTF-8

pandas           : 0.25.0
numpy            : 1.16.4
pytz             : 2019.1
dateutil         : 2.8.0
pip              : 19.0.3
setuptools       : 40.8.0
Cython           : None
pytest           : 5.0.1
hypothesis       : None
sphinx           : None
blosc            : None
feather          : None
xlsxwriter       : None
lxml.etree       : None
html5lib         : None
pymysql          : None
psycopg2         : None
jinja2           : None
IPython          : 7.6.1
pandas_datareader: None
bs4              : None
bottleneck       : None
fastparquet      : None
gcsfs            : None
lxml.etree       : None
matplotlib       : None
numexpr          : None
odfpy            : None
openpyxl         : None
pandas_gbq       : None
pyarrow          : None
pytables         : None
s3fs             : None
scipy            : None
sqlalchemy       : None
tables           : None
xarray           : None
xlrd             : None
xlwt             : None
xlsxwriter       : None
</details>

### PR DESCRIPTION
**Pull request by mroeschke**
- [x] closes #27484
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

### CODE REVIEW

File: doc/source/whatsnew/v0.25.1.rst [hunk 1]
```diff
@@ -56,7 +56,7 @@ Timezones
 
 Numeric
 ^^^^^^^
--
+- Bug in :class:`format` in which negative floating point complex numbers would raise an ``IndexError`` (:issue:`27484`)
```
Comment (jbrockmendel) [inline doc/source/whatsnew/v0.25.1.rst]: Is this the right class?

**Comment by jbrockmendel** [comment]
LGTM

**Comment by jreback** [comment]
lgtm. going to see if I can fix the CI on 0.25.1 first.

**Comment by jreback** [comment]
can you rebase this @mroeschke

**Comment by mroeschke** [comment]
Rebased and green.

**Review by WillAyd — APPROVED** [review]

**Comment by WillAyd** [comment]
Thanks @mroeschke

**Pull request merged**